# Regresion lineal sobre datos socio demográficos y de recursos de salud a nivel de condado de EE. UU. (2018-2019)

Instrucciones

Se han recopilado datos socio demográficos y de recursos de salud por condado en los Estados Unidos y queremos descubrir si existe alguna relación entre los recursos sanitarios y los datos socio demográficos.


In [8]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import math
# split para modelado
from sklearn.model_selection import train_test_split
# Scaled | Escalado
from sklearn.preprocessing import StandardScaler, MinMaxScaler
# Encoding | Codificación
from sklearn.preprocessing import LabelEncoder, OneHotEncoder, OrdinalEncoder
# To save models
import json
import pickle
# Feature Selection
from sklearn.feature_selection import f_classif, SelectKBest
# Modelado
from sklearn.linear_model import LogisticRegression
# Métricas
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix
# Optimizar
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import RandomizedSearchCV
# from tqdm import tqdm
import warnings
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
from sklearn.linear_model import Lasso

In [9]:
# Mostrar todas las filas
pd.set_option("display.max_rows", None)

# Mostrar todas las columnas
pd.set_option("display.max_columns", None)

# Ajustar el ancho de cada columna si es necesario
pd.set_option("display.max_colwidth", None)

# Aumentar el ancho total del display
pd.set_option("display.width", 0)

In [10]:
df = pd.read_csv('/workspaces/sarayruiz---Machine-Learning/data/raw/reg-demographic.csv', sep=',')
df.head()

fips  TOT_POP    0-9  0-9 y/o % of total pop  19-Oct  \
0  1001    55601   6787               12.206615    7637   
1  1003   218022  24757               11.355276   26913   
2  1005    24881   2732               10.980266    2960   
3  1007    22400   2456               10.964286    2596   
4  1009    57840   7095               12.266598    7570   

   10-19 y/o % of total pop  20-29  20-29 y/o % of total pop  30-39  \
0                 13.735364   6878                 12.370281   7089   
1                 12.344167  23579                 10.814964  25213   
2                 11.896628   3268                 13.134520   3201   
3                 11.589286   3029                 13.522321   3113   
4                 13.087828   6742                 11.656293   6884   

   30-39 y/o % of total pop  40-49  40-49 y/o % of total pop  50-59  \
0                 12.749771   7582                 13.636445   7738   
1                 11.564429  27338                 12.539102  29986   
2                 12.865239   3074                 12.354809   3278   
3                 13.897321   3038                 13.562500   3115   
4                 11.901798   7474                 12.921853   7844   

   50-59 y/o % of total pop  60-69  60-69 y/o % of total pop  70-79  \
0                 13.917016   5826                 10.478229   4050   
1                 13.753658  29932                 13.728890  20936   
2                 13.174712   3076                 12.362847   2244   
3                 13.906250   2545                 11.361607   1723   
4                 13.561549   6965                 12.041840   4931   

   70-79 y/o % of total pop   80+  80+ y/o % of total pop  White-alone pop  \
0                  7.284042  2014                3.622237            42660   
1                  9.602701  9368                4.296814           190301   
2                  9.018930  1048                4.212049            12209   
3                  7.691964   785                3.504464            17211   
4                  8.525242  2335                4.036999            55456   

   % White-alone  Black-alone pop  % Black-alone  \
0      76.725239            10915      19.630942   
1      87.285228            19492       8.940382   
2      49.069571            12042      48.398376   
3      76.834821             4770      21.294643   
4      95.878285              950       1.642462   

   Native American/American Indian-alone pop  % NA/AI-alone  Asian-alone pop  \
0                                        267       0.480207              681   
1                                       1684       0.772399             2508   
2                                        164       0.659137              113   
3                                         98       0.437500               53   
4                                        378       0.653527              185   

   % Asian-alone  Hawaiian/Pacific Islander-alone pop  % Hawaiian/PI-alone  \
0       1.224798                                   62             0.111509   
1       1.150343                                  146             0.066966   
2       0.454162                                   46             0.184880   
3       0.236607                                   26             0.116071   
4       0.319848                                   70             0.121024   

   Two or more races pop  % Two or more races  POP_ESTIMATE_2018  \
0                   1016             1.827305              55601   
1                   3891             1.784682             218022   
2                    307             1.233873              24881   
3                    242             1.080357              22400   
4                    801             1.384855              57840   

   N_POP_CHG_2018  GQ_ESTIMATES_2018  R_birth_2018  R_death_2018  \
0             158                455          11.8           9.6   
1            5403               2190          10.5          10.3   
2            -277     

In [11]:
df.columns


Index(['fips', 'TOT_POP', '0-9', '0-9 y/o % of total pop', '19-Oct',
       '10-19 y/o % of total pop', '20-29', '20-29 y/o % of total pop',
       '30-39', '30-39 y/o % of total pop',
       ...
       'COPD_number', 'diabetes_prevalence', 'diabetes_Lower 95% CI',
       'diabetes_Upper 95% CI', 'diabetes_number', 'CKD_prevalence',
       'CKD_Lower 95% CI', 'CKD_Upper 95% CI', 'CKD_number',
       'Urban_rural_code'],
      dtype='object', length=108)

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3140 entries, 0 to 3139
Columns: 108 entries, fips to Urban_rural_code
dtypes: float64(61), int64(45), object(2)
memory usage: 2.6+ MB


> ### Observaciones:
>
>- Tenemos en la base de datos 3140 filas y 108 columnas.
>- Solo dos columnas son Objetos. El resto son números.
>- 

## Limpieza de datos

In [13]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
5       False
6       False
7       False
8       False
9       False
10      False
11      False
12      False
13      False
14      False
15      False
16      False
17      False
18      False
19      False
20      False
21      False
22      False
23      False
24      False
25      False
26      False
27      False
28      False
29      False
30      False
31      False
32      False
33      False
34      False
35      False
36      False
37      False
38      False
39      False
40      False
41      False
42      False
43      False
44      False
45      False
46      False
47      False
48      False
49      False
50      False
51      False
52      False
53      False
54      False
55      False
56      False
57      False
58      False
59      False
60      False
61      False
62      False
63      False
64      False
65      False
66      False
67      False
68      False
69      False
70      False
71    

In [21]:
#Eliminar duplicados.
if df.duplicated().sum():
        df_inicio = df.drop_duplicates()
df_inicio.head()

NameError: name 'df_inicio' is not defined

In [15]:
df_inicio.drop(['fips', '0-9', '19-Oct', '20-29', '30-39', '40-49', '50-59', '60-69', '70-79', '80+', 'White-alone pop', 'Black-alone pop',
         'Native American/American Indian-alone pop', 'Asian-alone pop', 'Hawaiian/Pacific Islander-alone pop', 'Two or more races pop',
         'Population Aged 60+'], axis=1, inplace=True)
df_inicio.head()

NameError: name 'df_inicio' is not defined

> ### Observaciones:
>- Elimino esas columnas porque voy a trabajar con los %.